sdk/src/main/scala/io/horizen/certificatesubmitter/dataproof/CertificateDataWithKeyRotation.scala

```diff
@@ -6,28 +6,29 @@ import com.horizen.certnative.BackwardTransfer
 import io.horizen.cryptolibprovider.CryptoLibProvider
 import io.horizen.proof.SchnorrProof
 import io.horizen.proposition.SchnorrProposition
+import io.horizen.sc2sc.Sc2ScDataForCertificate
 import io.horizen.utils.BytesUtils
 import io.horizen.utxo.box.WithdrawalRequestBox
 
 import scala.collection.JavaConverters._
 
-
 case class CertificateDataWithKeyRotation(override val referencedEpochNumber: Int,
                                           override val sidechainId: Array[Byte],
                                           override val backwardTransfers: Seq[BackwardTransfer],
                                           override val endEpochCumCommTreeHash: Array[Byte],
+                                          override val sc2ScDataForCertificate: Option[Sc2ScDataForCertificate],
                                           override val btrFee: Long,
                                           override val ftMinAmount: Long,
                                           override val schnorrKeyPairs: Seq[(SchnorrProposition, Option[SchnorrProof])],
                                           schnorrKeysSignatures: SchnorrKeysSignatures,
                                           previousCertificateOption: Option[WithdrawalEpochCertificate],
                                           genesisKeysRootHash: Array[Byte]
                                          )
-  extends CertificateData(referencedEpochNumber, sidechainId, backwardTransfers, endEpochCumCommTreeHash, btrFee, ftMinAmount, schnorrKeyPairs) {
+  extends CertificateData(referencedEpochNumber, sidechainId, backwardTransfers, endEpochCumCommTreeHash, sc2ScDataForCertificate, btrFee, ftMinAmount, schnorrKeyPairs) {
 
   override def getCustomFields: Seq[Array[Byte]] = {
+    //TODO: use sc2ScDataForCertificate to add custom fields for sc2sc
```

Comment:
here we should create a seq of 32 elements, where the first element is used for `keysRootHash` and others 31 for whatever needs. In out case second and thirst is used for sc2sc (if enabled and activated)